src/Microsoft.Identity.Web.TokenAcquisition/TokenAcquisition.cs

```diff
@@ -358,10 +360,40 @@ private void LogAuthResult(AuthenticationResult? authenticationResult)
                 throw new ArgumentException(IDWebErrorMessage.ClientCredentialTenantShouldBeTenanted, nameof(tenant));
             }
 
+            // If using managed identity 
+            if (tokenAcquisitionOptions != null && tokenAcquisitionOptions.ManagedIdentity != null)
+            {
+                IManagedIdentityApplication miApplication;
+
+                if (tokenAcquisitionOptions.ManagedIdentity.ManagedIdentityType == ManagedIdentityType.UserAssigned &&
```

Comment:
do you see now why the ManagedIdentityType is not necessary?

tokenAcquisitionOptions.ManagedIdentity != null => it's a managed identity
- tokenAcquisitionOptions.ManagedIdentity.ClientId is not null => it's a user assigned managed identity
- tokenAcquisitionOptions.ManagedIdentity.ClientID is not => it's a system assigned managed identity

Reply:
I think that is easier to get wrong since if you miss the documentation saying system-assigned needs to be null and put in the ClientID of your application i.e. the one that correlates to your system your request will fail which feels like an anti-pattern to me. It is unintuitive that putting in the ClientID of your system application would cause system-assigned managed identity to fail.

Reply:
To be clear, when I mentionned caching the managed identity application, we'd probably want to use a concurrent dictionary of managed identity by ClientID (the empty client ID representing the system-assigned managed identity)